client/main.go

```diff
@@ -175,7 +176,17 @@ func sendNonStreamingRequests(client pb.ResponderClient,
 			return
 		default:
 			start := time.Now()
-			resp, err := client.Get(context.Background(),
+
+			var ctx context.Context
+			if clientTimeout != time.Duration(0) {
+				var cancel context.CancelFunc
+				ctx, cancel = context.WithTimeout(context.Background(), clientTimeout)
+				defer cancel()
```

Comment:
Since this is being called in a for loop, I _think_ that means that the `cancel` won't actually be called until the `sendNonStreamingRequests` method exits, after all requests are sent. Whereas we probably want to call `cancel` after each request. I think the easiest thing to do would be to move everything in this `default` block into a separate method, where `defer` will work properly.